### PR TITLE
`Development`: Add JavaDocs to UtilServices and Factories [6]

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to the helper functions

### Description

Added documentation to following files:
- [x] exercise/modelingexercise/ModelingExerciseFactory.java
- [x] exercise/modelingexercise/ModelingExerciseUtilService.java

### Steps for Testing
_JavaDoc changes only_

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseUtilService.java [hunk 1]
```diff
@@ -99,21 +101,21 @@ public Course addCourseWithOneModelingExercise(String title) {
...
-        Course course = CourseFactory.generateCourse(null, pastTimestamp, futureFutureTimestamp, new HashSet<>(), "tumuser", "tutor", "editor", "instructor");
-        ModelingExercise modelingExercise = ModelingExerciseFactory.generateModelingExercise(pastTimestamp, futureTimestamp, futureFutureTimestamp, DiagramType.ClassDiagram,
-                course);
-        modelingExercise.setTitle(title);
-        course.addExercises(modelingExercise);
-        courseRepo.save(course);
-        exerciseRepo.save(modelingExercise);
-        return course;
-    }
-
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseUtilService.java:116]: One usage replaced with _Course addCourseWithOneModelingExercise(String title)_

File: src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseUtilService.java [hunk 2]
```diff
@@ -318,26 +407,17 @@ public Result addModelingAssessmentForSubmission(ModelingExercise exercise, Mode
...
-        ModelingExercise modelingExercise = ModelingExerciseFactory.generateModelingExercise(pastTimestamp, futureTimestamp, futureFutureTimestamp, DiagramType.ClassDiagram,
-                course);
-        modelingExercise.setGradingInstructions("Grading instructions");
-        modelingExercise.getCategories().add("Modeling");
-        modelingExercise = modelingExerciseRepository.save(modelingExercise);
-        course.addExercises(modelingExercise);
-        return courseRepo.save(course);
-
-    }
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseUtilService.java:342]: Unused

**Review by laadvo — APPROVED** [review]
Looks good, I only have two small suggestions

File: src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseFactory.java [hunk 3]
```diff
@@ -43,6 +68,13 @@ public static ModelingExercise generateModelingExerciseForExam(DiagramType diagr
         return modelingExercise;
     }
 
+    /**
+     * Generates an empty ApollonDiagram with the given DiagramType.
```
[the remark this answers is not in the document]
Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/exercise/modelingexercise/ModelingExerciseFactory.java:72]: Done. I added information about the jsonRepresentation that is not set

**Pull request merged**